podcasts/SwiftUI/ContentSizeReader.swift

```diff
@@ -0,0 +1,55 @@
+import SwiftUI
+
+/// Calculates the size of the containing view and returns it
+/// This allows you to get easily get the size of the view without needing all the boilerplate for `GeometryReader`
+struct ContentSizeReader<Content: View>: View {
+    private var content: (CGSize?) -> Content
+
+    init(@ViewBuilder _ content: @escaping (CGSize?) -> Content) {
+        self.content = content
+    }
+
+    @State private var size: CGSize?
+
+    var body: some View {
+        // If we've calculated the size, then just return it and the view
+        if let size {
+            content(size)
+        } else {
+            content(nil).background(
```

Comment:
I don't think using `GeometryReader` is too problematic in terms of boilerplate code by doing `reader.size.width` or `reader.size.height`.

I also don't see a problem using this wrapper, I just want to point one key difference: this wrapper forces the view to render twice (because of the `@State size`).

I'm not sure if there's any cost in terms of CPU — I guess not because SwiftUI is smart enough to only re-render views if something changed — but if someone uses `ContentSizeReader` together with a `.onAppear` this will get called twice, which I think is an unintended behavior.